### PR TITLE
Add config for which host sytest should bind ports on.

### PR DESCRIPTION
**Pull request by NegativeMjark**
This can be used to run multiple instances of sytest on the same machine
by supplying different 127.*.*.* addresses for each instance.

### CODE REVIEW

**Comment by erikjohnston** [comment]
Would be worth adding a BIND_HOST env var to `jenkins/sytest*.sh`?

File: lib/SyTest/Homeserver.pm [hunk 1]
```diff
@@ -97,12 +97,9 @@ sub await_connectable
...
       $loop->connect(
-         addr => {
-            family   => "inet",
-            socktype => "stream",
-            port     => $port,
-            ip       => "127.0.0.1",
-         }
+         host => $host,
+         service => $port,
+         socktype => "stream",
```
Comment (leonerd) [inline lib/SyTest/Homeserver.pm]: Arrow alignment

File: tests/05synapse.pl [hunk 2]
```diff
@@ -61,14 +61,14 @@ END
...
+            "https://$BIND_HOST:$secure_port" :
+            "http://$BIND_HOST:$unsecure_port";
 
-         my $info = ServerInfo( "localhost:$secure_port", $location );
+         my $info = ServerInfo( "$BIND_HOST:$secure_port", $location );
 
          my $synapse = SyTest::Homeserver::Synapse->new(
             synapse_dir   => $SYNAPSE_ARGS{directory},
-            hs_dir        => abs_path( "localhost-$idx" ),
+            hs_dir        => abs_path( "$BIND_HOST-$idx" ),
```
Comment (leonerd) [inline tests/05synapse.pl]: You don't want to change this one (as per IRL)

File: tests/07id-server.pl [hunk 3]
```diff
@@ -15,7 +15,7 @@ sub id_server_fixture
          $loop->add( $id_server );
 
          $id_server->listen(
-            host    => "localhost",
+            host    => "$BIND_HOST",
```
Comment (leonerd) [inline tests/07id-server.pl]: Don't bother quoting this; just

    host => $BIND_HOST

File: tests/50federation/00prepare.pl [hunk 4]
```diff
@@ -21,7 +21,7 @@
       require IO::Async::SSL;
 
       $inbound_server->listen(
-         host    => "localhost",
+         host    => "$BIND_HOST",
```
Comment (leonerd) [inline tests/50federation/00prepare.pl]: Ditto quote

File: run-tests.pl [hunk 5]
```diff
@@ -55,6 +55,8 @@
 
 our $WANT_TLS = 1;  # This is shared with the test scripts
 
+our $BIND_HOST = "localhost";
```
Comment (leonerd) [inline run-tests.pl]: I'd group these "shared with test scripts" variables into a block without blank lines, and move the comment to the top of the block, now that there's several.

**Pull request merged**